src/components/map/UserCard.tsx

```diff
@@ -82,16 +84,21 @@ const UserCard = ({ user, onClose }: UserCardProps) => {
       }}>
       {/* User Header */}
       <View style={UserCardStyles.userHeaderContainer}>
-        <View
-          style={{
-            ...UserCardStyles.userProfilePic,
-            borderColor: statusColor,
-          }}>
+        {/* Conditionally render user profile pic or initials */}
+        {user.imgUrlProfileLarge ? (
+          <Image
+            source={{ uri: user.imgUrlProfileLarge }}
+            style={{
+              ...UserCardStyles.userProfilePic,
+              borderColor: statusColor,
+            }}
+          />
+        ) : (
           <Text style={UserCardStyles.userProfilePicText}>
             {user.firstName[0]}
             {user.lastName[0]}
           </Text>
-        </View>
+        )}
```

Comment:
i would like to make this a util function as well or component or something

Reply:
TODO for after backend added the endpoint for querying many users at once